oaknational/Oak-Web-Application
# [ENG-132] feat: enable A/B testing of landing pages via the CMS

## Description
- Landing pages are now entirely server rendered (no static, no ISR). This is to avoid a bad UX where the control variant is shown until the client-side JS takes over and then a variant is re-rendered, and because we can only get the users cookies via SSR
- If a landing page with an experiment is encountered, the user will be redirected to one of the variants. These were initially all rendered on the same URL, but it was swapped to redirecting because of complications with hubspot tracking
- Users that have previously consented and therefor have a posthog cookie will be shown the variant posthog decides, entirely anonymous users will be sent to a randomly selected control or variant page
- Landing pages without experiments should be exactly the same, just SSR'd

Known issue: Currently if a user has no cookies set they will see a random variant. If they then accept cookies and return to the experiment page, they will potentially see a different variant. This is a tradeoff agreed with Simon H.

TODO: Unset the `SANITY_DATASET_TAG`, which is currently set to `ab-test` for this branch specifically

## Issue(s)

Fixes #

## How to test
As an identified user:
1. Go to the [deploy preview homepage](https://deploy-preview-1766--oak-web-application.netlify.thenational.academy)
2. Accept cookies
3. Go to the A/B tested landing page[/lp/ab-test](https://deploy-preview-1766--oak-web-application.netlify.thenational.academylp/ab-test)
4. You should be redirected to a variant page. Each time you return to the above link you're redirected to the same variant.

As an anonymous user
1. Clear cookies & site data
2. Go to the A/B tested landing page[/lp/ab-test](https://deploy-preview-1766--oak-web-application.netlify.thenational.academylp/ab-test)
3. Do not accept cookies
4. Each time you return to the above link you should be redirected to a random variant

## Checklist
